### ISSUE TITLE
Can't initialize NVML

### ISSUE THREAD

**Issue by rjtupas** [remsky/Kokoro-FastAPI#527]
**Describe the bug**
On docker container start, receive the following: "/app/.venv/lib/python3.12/site-packages/torch/cuda/__init__.py:829: UserWarning: Can't initialize NVML "

**Screenshots or console output**
N/A

**Branch / Deployment used**
docker run --gpus all -p 8880:8880 ghcr.io/remsky/kokoro-fastapi-gpu:latest

**Operating System**
Gentoo Linux
GPU 0: NVIDIA GeForce RTX 4060 Laptop GPU (UUID: GPU-3cb0bb58-f79f-2324-1cb9-f1e49b84af89)

**Additional context**
1. nvidia-smi works
2. Ran the following code using python3.12 (same as koko-fastapi-gpu container):

import subprocess
import torch

if __name__ == '__main__':
    print('Testing NVIDIA SMI')
    try:
        # Run the nvidia-smi command
        result = subprocess.run(['nvidia-smi'], capture_output=True, text=True, check=True)
        # Print the output
        print("Result", result)
        print("Result stdout: ", result.stdout)
    except subprocess.CalledProcessError as e:
        print(f"subprocess.CalledProcessError occurred: {e}")
    print('Testing CUDA')
    try:
        print("CUDA CHECK ---------------------")
        print('CUDA available running "torch.cuda.is_available()":', torch.cuda.is_available())
        print('CUDA device count:"torch.cuda.device_count()"', torch.cuda.device_count())
        print('CUDA device name:', torch.cuda.get_device_name(0))
        if torch.cuda.is_available():
            for idx in range(torch.cuda.device_count()):
                name = torch.cuda.get_device_name(idx)
                cap = torch.cuda.get_device_capability(idx)
                mem = torch.cuda.get_device_properties(idx).total_memory/1e9
            print(f"{name}  (sm{cap[0]}{cap[1]})  {mem:.1f} GB")
    except Exception as e:
        print(f"Error occurred: {e}")

Result stdout:  Mon Sep  7 13:05:11 2026       
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 595.99.02              Driver Version: 595.99.02      CUDA Version: 13.2     |
+-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA GeForce RTX 4060 ...    Off |   00000000:01:00.0  On |                  N/A |
| N/A   46C    P8              1W /   80W |      71MiB /   8188MiB |      0%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+

+-----------------------------------------------------------------------------------------+
| Processes:                                                                              |
|  GPU   GI   CI              PID   Type   Process name                        GPU Memory |
|        ID   ID                                                               Usage      |
|=========================================================================================|
|    0   N/A  N/A            4512      G   /usr/bin/gnome-shell                     42MiB |
|    0   N/A  N/A           12976      G   /app/bin/gnome-calendar                   2MiB |
+-----------------------------------------------------------------------------------------+

Testing CUDA
CUDA CHECK ---------------------
CUDA available running "torch.cuda.is_available()": True
CUDA device count:"torch.cuda.device_count()" 1
CUDA device name: NVIDIA GeForce RTX 4060 Laptop GPU
NVIDIA GeForce RTX 4060 Laptop GPU  (sm89)  8.2 GB

**Comment by rjtupas** [comment]
Forgot to add:
3. Also ran successfully: docker run --rm --gpus all nvidia/cuda:12.0.0-base-ubuntu22.04 nvidia-smi
 
4.  From docker logs, also received: 
/app/.venv/lib/python3.12/site-packages/torch/nn/modules/rnn.py:123: UserWarning: dropout option adds dropout after all but last recurrent layer, so non-zero dropout expects num_layers greater than 1, but got dropout=0.2 and num_layers=1
  warnings.warn(
/app/.venv/lib/python3.12/site-packages/torch/nn/utils/weight_norm.py:144: FutureWarning: `torch.nn.utils.weight_norm` is deprecated in favor of `torch.nn.utils.parametrizations.weight_norm`.
  WeightNorm.apply(module, name, dim)

**Comment by remsky** [comment]
Hey @rjtupas 

In the startup banner, is it showing
>Model warmed up on cuda: ...
> CUDA: True

If so it should still be working fine on GPU inference. If not, my first bet would be that its the linux permissions. You could try the following to see if adding the `video` perms for the container helps, let me know if not:

```bash
docker run --gpus all --group-add $(stat -c %g /dev/nvidiactl) -p 8880:8880 ghcr.io/remsky/kokoro-fastapi-gpu:latest
```

The warning is just torch attempting to use NVML to count GPUs at startup (why that probe is failing I'm not sure, could be related to the specific host driver or setup), but if/when the probe fails it falls back to counting GPUs via the CUDA runtime which works as well. 

---

On 3. 

>  UserWarning: dropout option adds ...
> FutureWarning: torch.nn.utils.weight_norm ...

This is from the base Kokoro library; a prexisting warning we pass-through, and it doesn't cause any issues.

**Comment by rjtupas** [comment]
1. The startup banner shows:
Model warmed up on cuda: kokoro_v1
CUDA: False

2. Ran the command provided.
Model warmed up on cuda: kokoro_v1
CUDA: True <<<===== Yea!!!

3. Understand the other warning do not cause any issues. Thanks!

**Comment by remsky** [comment]
That's great. I'll update the [troubleshooting guide](https://github.com/remsky/Kokoro-FastAPI/blob/master/docs/troubleshooting.md) for this case. Glad its working
